src/compute/src/render/reduce.rs

```diff
@@ -1476,17 +1471,33 @@ where
                             | (
                                 AggregateFunc::SumUInt32,
                                 Accum::SimpleNumber { accum, .. },
-                            )
+                            ) => {
+                                if !accum.is_negative() {
+                                    // Our semantics of overflow are not clearly articulated wrt.
+                                    // unsigned vs. signed types (#17758). We adopt an unsigned
+                                    // wrapping behavior to match what we do above for signed types.
+                                    // TODO(vmarcos): remove potentially dangerous usage of `as`.
+                                    #[allow(clippy::as_conversions)]
+                                    Datum::UInt64(*accum as u64)
```

Comment:
```suggestion
                                    Datum::UInt64(accum.try_into().unwrap_or(0))
```

Reply:
I think that this takes the opinion that we should not wrap around, but rather return zero? I'd like to leave a decision for such a deviation in behavior wrt. signed types to be evaluated as part of #17758. Perhaps we'd prefer to turn these into errors uniformly everywhere, but this exceeds the scope of this PR.

Reply:
Makes sense